src/internal/stream.settings.cpp

```diff
@@ -0,0 +1,539 @@
+
+    return ZarrError_Success;
+}
+
+ZarrError
+ZarrStreamSettings_set_multiscale(ZarrStreamSettings* settings,
+                                  uint8_t multiscale)
+{
+    EXPECT_VALID_ARGUMENT(settings, "Null pointer: settings");
+
+    settings->multiscale = multiscale > 0;
+    return ZarrError_Success;
+}
+
+/* Getters */
+const char*
+ZarrStreamSettings_get_store_path(const ZarrStreamSettings* settings)
+{
+    SETTINGS_GET_STRING(settings, store_path);
+}
+
+const char*
+ZarrStreamSettings_get_s3_endpoint(const ZarrStreamSettings* settings)
+{
+    SETTINGS_GET_STRING(settings, s3_endpoint);
+}
+
+const char*
+ZarrStreamSettings_get_s3_bucket_name(const ZarrStreamSettings* settings)
+{
+    SETTINGS_GET_STRING(settings, s3_bucket_name);
+}
+
+const char*
+ZarrStreamSettings_get_s3_access_key_id(const ZarrStreamSettings* settings)
+{
+    SETTINGS_GET_STRING(settings, s3_access_key_id);
+}
+
+const char*
+ZarrStreamSettings_get_s3_secret_access_key(const ZarrStreamSettings* settings)
+{
+    SETTINGS_GET_STRING(settings, s3_secret_access_key);
+}
+
+const char*
+ZarrStreamSettings_get_external_metadata(const ZarrStreamSettings* settings)
+{
+    SETTINGS_GET_STRING(settings, external_metadata);
+}
+
+ZarrDataType
+ZarrStreamSettings_get_data_type(const ZarrStreamSettings* settings)
+{
+    if (!settings) {
+        LOG_WARNING("Null pointer: settings. Returning DataType_uint8.");
+        return ZarrDataType_uint8;
+    }
+    return static_cast<ZarrDataType>(settings->dtype);
+}
+
+ZarrCompressor
+ZarrStreamSettings_get_compressor(const ZarrStreamSettings* settings)
+{
+    if (!settings) {
+        LOG_WARNING("Null pointer: settings. Returning ZarrCompressor_None.");
+        return ZarrCompressor_None;
+    }
+    return static_cast<ZarrCompressor>(settings->compressor);
+}
+
+ZarrCompressionCodec
+ZarrStreamSettings_get_compression_codec(const ZarrStreamSettings* settings)
+{
+    if (!settings) {
+        LOG_WARNING(
+          "Null pointer: settings. Returning ZarrCompressionCodec_None.");
+        return ZarrCompressionCodec_None;
+    }
+    return static_cast<ZarrCompressionCodec>(settings->compression_codec);
+}
+
+uint8_t
+ZarrStreamSettings_get_compression_level(const ZarrStreamSettings* settings)
+{
+    if (!settings) {
+        LOG_WARNING("Null pointer: settings. Returning 0.");
+        return 0;
+    }
+    return settings->compression_level;
+}
+
+uint8_t
+ZarrStreamSettings_get_compression_shuffle(const ZarrStreamSettings* settings)
+{
+    if (!settings) {
+        LOG_WARNING("Null pointer: settings. Returning 0.");
+        return 0;
+    }
+    return settings->compression_shuffle;
+}
+
+size_t
+ZarrStreamSettings_get_dimension_count(const ZarrStreamSettings* settings)
+{
+    if (!settings) {
+        LOG_WARNING("Null pointer: settings. Returning 0.");
+        return 0;
+    }
+    return settings->dimensions.size();
+}
+
+ZarrError
+ZarrStreamSettings_get_dimension(const ZarrStreamSettings* settings,
```

Comment:
- I think there needs to be a distinction in the name between an accessor function that returns a value and a function that is used to assign values into output parameters.
- The signature of this function reinforces the need for a struct that represents a "dimension", instead of passing individual properties.